GatherPress/gatherpress
# Gatherpress playwright setup

Below are the changes done as part of this PR

- Created playwright framework and added dependencies
- Created first test - 
 As admin login into gatherPress:  The Event menu item should be preloaded after clicking Add New button
- Created reusable helper file
- Created .env file to store config information. For this test we are storing password information
- Added npm script to run the test in headless mode in package.json


# Steps to run playwright tests
1. Run `npm i`
2. Configure password of the login user in .env file (located at root of the project)
WP_ADMIN_PASSWORD="your password"
3. To run the playwright test - From project root folder execute `npm run tests:e2e-all` 
4. If you see any browser related issue then you may have to execute `npx playwright install` to update the playwright browsers
5. After executing step 2 you can see the reports by running `npx playwright show-report`
6. You should see tests executed in Chrome, webkit (Safari), Edge and Firefox browsers


Screenshot of the sample test
![Test_Screenshot](https://github.com/GatherPress/gatherpress/assets/10412309/b3bcaf49-145f-4389-a81c-216895fb41eb)

Screenshot of the sample video
[Test_video.webm](https://github.com/GatherPress/gatherpress/assets/10412309/452010af-ed00-432f-b8fb-4bd3081c47c7)

In the sample test we are verifying preloaded blocks in the event post through screenshot
